From 798df4465978b16bafa53d760214ec21f81ac8eb Mon Sep 17 00:00:00 2001
From: Jan Tulak <jtulak@redhat.com>
Date: Thu, 21 Sep 2017 14:43:34 +0200
Subject: [PATCH 4/5] tests: fix bashtests/008-btrfs-remove.sh

Fix an issue with btrfs test: btrfs doesn't allow to delete last two
devices once it creates a raid. So, don't try to remove all, remove only
some to see if removal works.

Signed-off-by: Jan Tulak <jtulak@redhat.com>
---
 tests/bashtests/008-btrfs-remove.sh | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/tests/bashtests/008-btrfs-remove.sh b/tests/bashtests/008-btrfs-remove.sh
index 4e40e99..705a76d 100755
--- a/tests/bashtests/008-btrfs-remove.sh
+++ b/tests/bashtests/008-btrfs-remove.sh
@@ -62,10 +62,13 @@ btrfs filesystem show
 ssm add $TEST_DEVS
 btrfs filesystem show
 check btrfs_fs_field $SSM_BTRFS_DEFAULT_POOL dev_count $DEV_COUNT
-ssm -f remove $TEST_DEVS
+# Skip first few devices, because btrfs is now in a raid mode
+# and won't allow us to delete the last few devices.
+# We only want to see if removal works, so deleting only some is ok.
+ssm -f remove $(echo $TEST_DEVS | cut -d' ' -f5-)
 btrfs filesystem show
 ssm list
-check btrfs_fs_field $SSM_BTRFS_DEFAULT_POOL dev_count 1
+check btrfs_fs_field $SSM_BTRFS_DEFAULT_POOL dev_count 4
 ssm -f remove --all
 
 # Remove multiple things
-- 
2.17.1

